src/typeahead/typeahead.js

```diff
@@ -170,6 +169,28 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.position', 'ui.bootstrap
         });
       };
 
+      // bind events only if appendToBody params exist - performance feature
+      if ( appendToBody ) {
+        angular.element( $window ).bind( 'resize', fireRecalculating );
+        angular.element( document.body ).bind( 'scroll', fireRecalculating );
```

Comment:
These bindings should be throttled.

Also you can use $document.find('body') instead.